File: task_1/cpp/1(1).cpp
#include <iostream>
#include <vector>
#include <cstdlib>   // для rand() и srand()
#include <ctime>     // для time()

using namespace std;

int main() {
    int N;
    cout << "Введите порядок квадратной матрицы N: ";
    cin >> N;

    // Инициализация генератора случайных чисел
    srand(static_cast<unsigned>(time(0)));

    // Создание и инициализация матрицы
    vector<vector<int>> matrix(N, vector<int>(N));
    cout << "\nСгенерированная матрица:\n";
    for (int i = 0; i < N; ++i) {
        for (int j = 0; j < N; ++j) {
            matrix[i][j] = rand() % 201 - 100; // от -100 до 100
            cout << matrix[i][j] << "\t";
        }
        cout << endl;
    }

    // Поиск строки с наибольшим числом положительных элементов
    int maxPositives = 0;
    int maxRowIndex = 0;

    for (int i = 0; i < N; ++i) {
        int positiveCount = 0;
        for (int j = 0; j < N; ++j) {
            if (matrix[i][j] > 0) {
                ++positiveCount;
            }
        }
        if (positiveCount > maxPositives) {
            maxPositives = positiveCount;
            maxRowIndex = i;
        }
    }

    // Инициализация нового массива найденной строкой
    vector<int> resultRow = matrix[maxRowIndex];

    // Вывод результата
    cout << "\nСтрока с наибольшим количеством положительных чисел (номер " << maxRowIndex << "):\n";
    for (int val : resultRow) {
        cout << val << " ";
    }
    cout << endl;

    return 0;
}

File: task_1/go/1(1).go
package main

import (
	"fmt"
	"math/rand"
	"time"
)

func main() {
	var N int
	fmt.Print("Введите порядок квадратной матрицы N: ")
	fmt.Scan(&N)

	// Инициализация генератора случайных чисел
	rand.Seed(time.Now().UnixNano())

	// Создание и инициализация матрицы
	matrix := make([][]int, N)
	fmt.Println("\nСгенерированная матрица:")
	for i := 0; i < N; i++ {
		matrix[i] = make([]int, N)
		for j := 0; j < N; j++ {
			matrix[i][j] = rand.Intn(201) - 100 // от -100 до 100
			fmt.Printf("%d\t", matrix[i][j])
		}
		fmt.Println()
	}

	// Поиск строки с наибольшим числом положительных элементов
	maxPositives := 0
	maxRowIndex := 0

	for i := 0; i < N; i++ {
		positiveCount := 0
		for j := 0; j < N; j++ {
			if matrix[i][j] > 0 {
				positiveCount++
			}
		}
		if positiveCount > maxPositives {
			maxPositives = positiveCount
			maxRowIndex = i
		}
	}

	// Инициализация нового массива найденной строкой
	resultRow := matrix[maxRowIndex]

	// Вывод результата
	fmt.Printf("\nСтрока с наибольшим количеством положительных чисел (номер %d):\n", maxRowIndex)
	for _, val := range resultRow {
		fmt.Printf("%d ", val)
	}
	fmt.Println()
}
